### PR TITLE
[Remove Stream/Optional: Phase 4] LPS-182656 Remove Stream and optional usages for modules: dynamic-data-mapping-data-provider-impl、dynamic-data-mapping-form-renderer

### PR DESCRIPTION
**Pull request by W2jx**
resend

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4363

### CODE REVIEW

**Comment by Kvale1733** [comment]
@W2jx No regressions.

**Comment by W2jx** [comment]
no conflicts(4/27 11:15)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java [hunk 1]
```diff
@@ -523,18 +512,14 @@ private String _getPortalCacheKey(
 		String ddmDataProviderId, List<KeyValuePair> keyValuePairs,
 		String url) {
 
-		Stream<KeyValuePair> stream = keyValuePairs.stream();
-
 		return StringBundler.concat(
 			ddmDataProviderId, StringPool.AT, url, StringPool.QUESTION,
-			stream.sorted(
-			).map(
+			com.liferay.petra.string.StringUtil.merge(
```
Comment (dantewang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java:517]: @W2jx I missed this one in last review. Please make it a for-loop with StringBundler.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@W2jx Please resend, thanks!

**Pull request closed without merging**